Upload of profile image to S3

nravich2 opened the pull request.

erickpeirson committed:
added iterator to queryset chain in rest viewsets

erickpeirson committed:
fixed bug with iterator

erickpeirson committed:
added pagination to REST views

erickpeirson committed:
hotfix: VGNWB-143 added pagination to rest interface

erickpeirson committed:
hotfix baselocation in text view

erickpeirson committed:
proper filtering of relations in text annotation view

erickpeirson committed:
proper filtering of relations in text annotation view

erickpeirson committed:
hotfix concept results

Commit:
[VGNWB130]: Removed upload_to field in VogonUser Model and changed it to URL field. Modified view.py to save the URL to the model's imagefile field. Modified forms.py to add clean methods for fields validation

Commit:
Merge branch 'master' of https://github.com/diging/vogon-web into VGNWB-130

Commit:
[VGNWB130]:Modified view.py to have the image from S3 if profile image is already saved. Else, default image will be provided. Modififed settings.html to add name tags to bring in the data to POST method. Added field validations in User Change Form

erickpeirson committed:
Merge branch 'develop' into VGNWB-130

erickpeirson commented on annotations/views.py:
```diff
@@ -883,4 +929,35 @@ def concept_details(request, conceptid):
     return HttpResponse(template.render(context))
 
 
+def sign_s3(request):
```
This needs to be protected; use the [``@login_required`` decorator](https://docs.djangoproject.com/en/1.9/topics/auth/default/#django.contrib.auth.decorators.login_required).

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,121 @@
+              </div>
+              <div class="form-group">
+              <label for="imagefile">Please upload an image</label>
+              <input type="file" id="file_input"/>
```
This still needs to be moved outside of the form.

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,121 @@
+       var files = document.getElementById("file_input").files;
+       var file = files[0];
+       if(file == null){
+           alert("No file selected.");
```
Let's not use an alert; updating the profile image is enough.

erickpeirson replied:
+1

erickpeirson committed:
Merge branch 'VGNWB-82' into VGNWB-130

erickpeirson commented:
Awesome! Just a couple of quick tweaks, and this is good to go :-)

erickpeirson committed:
Merge branch 'master' into develop

Commit:
Merge branch 'develop' of https://github.com/diging/vogon-web into VGNWB-130

jasdeepbhalla committed:
[VGNWB-130] worked on PR comments

Commit:
Merge branch 'VGNWB-130' of https://github.com/diging/vogon-web into VGNWB-130

Commit:
VGNWB-130:Added default image in S3 and the reference of it in settings file.

nravich2 commented:
Worked on your comments.
Added the default image to S3 and in config file , added the URL so the default URL gets saved into the model without validation errors.

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,131 @@
+           }
+       };
+       xhr.onerror = function() {
+           alert("Could not upload file."); 
```
No alerts; use some other visual cue

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,131 @@
+       Function to carry out the actual PUT request to S3 using the signed request from the Python app.
+   */
+   function upload_file(file, signed_request, url){
+       var xhr = new XMLHttpRequest();
```
We could start the spinner here

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,131 @@
+       xhr.open("PUT", signed_request);
+       xhr.setRequestHeader('x-amz-acl', 'public-read');
+       xhr.onload = function() {
+           if (xhr.status === 200) {
```
And end the spinner here

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,131 @@
+               document.getElementById("defaultImage").value = url;
+           }
+       };
+       xhr.onerror = function() {
```
And also end the spinner here

erickpeirson commented on vogon/local_settings.py:
```diff
@@ -169,3 +171,10 @@
+# AWS Access Key and Secret Key credentials
+
+AWS_ACCESS_KEY     = 'AKIAIAM452M3MX7RRNGA'
+AWS_SECRET_KEY = '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'
```
Yikes, no secrets in repo!!!!!

erickpeirson commented on vogon/local_settings.py:
```diff
@@ -169,3 +171,10 @@
+AWS_ACCESS_KEY     = 'AKIAIAM452M3MX7RRNGA'
+AWS_SECRET_KEY = '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'
+S3_BUCKET = 'vogonweb-test'
+DEFAULT_USER_IMAGE = 'https://s3-us-west-2.amazonaws.com/vogonweb-test/defaultprofile.png'
```
Make sure this gets into `heroku_settings`, too

erickpeirson commented on vogon/urls.py:
```diff
@@ -71,8 +71,8 @@
-    url(r'^accounts/profile/$', views.dashboard, name='dashboard'),
-    url(r'^$', views.home)
+    url(r'^$', views.home),
+    url(r'^sign_s3$', views.sign_s3)
```
Don't forget to give this URL a name, so that we can load it in the template

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,131 @@
+   */
+   function get_signed_request(file){
+       var xhr = new XMLHttpRequest();
+       xhr.open("GET", "/sign_s3?file_name="+file.name+"&file_type="+file.type);
```
Once you give the `sign_s3` route a name in `urls.py` (see line comment in that file), you should use reverse URL resolution here, e.g.:

``` python
{% url "sign_s3" %}
```

erickpeirson commented:
This is really coming along! Still a few tweaks here and there, but overall very very close. Great work! :-)

jasdeepbhalla committed:
[VGNWB-130] Made changes as per the comments regarding the accesskeys

jasdeepbhalla committed:
[VGNWB-130] updated sign_s3

jasdeepbhalla committed:
[VGNWB-130] Added bootstrap based alert

Commit:
[VGNWB130]:Added bootstrap alerts and removed javascript alerts

jasdeepbhalla committed:
[VGNWB-130] Adding spinner to the settings page

Commit:
[VGNWB-130]: Adding spinner element

erickpeirson commented on annotations/templates/annotations/settings.html:
```diff
@@ -1,71 +1,170 @@
+      start upload procedure by asking for a signed request from the app.
+   */
+   function init_upload(){   
+       spinner=loadSpinner('spinnerContainer');          
```
You guys are overthinking this! ;-) All you need is this:

``` javascript
$('#spinnerContainer').spin(opts);
```

The `loadSpinner` function is overkill, and you don't need to instantiate the `Spinner` class (where did you get that, anyway?). Make sure that `opts` is global, and then literally all you have to do is call the `spin()` method of the jQuery selector.

jasdeepbhalla committed:
[VGNWB-130] Added Spinner

Commit:
[VGNWB-130]: Positioning the spinner element on top of the image.

erickpeirson commented:
Great work!
